presto-main/src/main/java/io/prestosql/metadata/AggregationFunctionMetadata.java

```diff
@@ -13,18 +13,39 @@
  */
 package io.prestosql.metadata;
 
+import io.prestosql.spi.type.StandardTypes;
 import io.prestosql.spi.type.TypeSignature;
+import io.prestosql.spi.type.TypeSignatureParameter;
 
+import java.util.Arrays;
 import java.util.Optional;
 
 import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.util.Objects.requireNonNull;
 
 public class AggregationFunctionMetadata
 {
     private final boolean orderSensitive;
     private final Optional<TypeSignature> intermediateType;
 
+    public AggregationFunctionMetadata(boolean orderSensitive, TypeSignature... intermediateTypes)
+    {
+        this.orderSensitive = orderSensitive;
+
+        if (intermediateTypes.length == 0) {
+            intermediateType = Optional.empty();
+        }
```

Comment:
How can an aggregation not have an intermediate type?

Reply:
non-decomposable aggregations do not have aggregates